modules/currency-l0/src/main/scala/org/tessellation/currency/l0/node/L0NodeContext.scala

```diff
@@ -1,16 +1,29 @@
 package org.tessellation.currency.l0.node
 
 import org.tessellation.currency.dataApplication.L0NodeContext
+import org.tessellation.currency.schema.currency
+import org.tessellation.currency.schema.currency.{CurrencyIncrementalSnapshot, CurrencySnapshotInfo}
 import org.tessellation.schema.{GlobalIncrementalSnapshot, GlobalSnapshotInfo}
 import org.tessellation.sdk.domain.snapshot.storage.LastSnapshotStorage
 import org.tessellation.security.{Hashed, SecurityProvider}
 
 object L0NodeContext {
   def make[F[_]: SecurityProvider](
-    lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo]
-  ): L0NodeContext[F] = new L0NodeContext[F] {
-    def getLastGlobalSnapshot: F[Option[Hashed[GlobalIncrementalSnapshot]]] = lastGlobalSnapshotStorage.get
+    lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
+    lastSnapshotStorage: LastSnapshotStorage[F, CurrencyIncrementalSnapshot, CurrencySnapshotInfo]
+  ): L0NodeContext[F] = {
+    val sp = SecurityProvider[F]
 
-    def securityProvider: SecurityProvider[F] = SecurityProvider[F]
+    new L0NodeContext[F] {
+      def getLastGlobalSnapshot: F[Option[Hashed[GlobalIncrementalSnapshot]]] = lastGlobalSnapshotStorage.get
+
+      def securityProvider: SecurityProvider[F] = sp
+
+      def getLastCurrencySnapshot: F[Option[Hashed[currency.CurrencyIncrementalSnapshot]]] = lastSnapshotStorage.get
```

Comment:
I think you can use direct import to `CurrencyIncrementalSnapshot` and `CurrencySnapshotInfo`